### DISCUSSION TITLE
How to estimate general memory requirements?

### DISCUSSION THREAD

**Discussion by sergeystoma** [duckdb/duckdb#9330] [General]
Trying to understand how to deploy duck-based solution and figure out system requirements - or limits of when things would break. Or how that "larger than memory" is supposed to work or where it would apply. 

* Are there any good writeups somewhere already? 
* Like, is the dataset supposed to fit into RAM? 
* Is the *compressed* dataset supposed to fit in RAM? 
* Is the *resultset* supposed to fit? 
 
So many questions 😄 

Trying to play on an intentionally limited (512MB RAM) VM; I've got 20GB of raw JSON data, which compresses into a 350MB parquet file. 

* `free -h` -> `total 380Mi used 189Mi free 89Mi`
* `create table test as select * from '1.parquet';` -> `Killed` (by OOM), so copying data needs RAM, ok
* `select <... some group by query...> from parquet_scan(*.parquet)` (16 copies of the same 350MB file) -> woah, that works!
* `set memory_limit="10MB"; set temp_directory="..."` -> rerun the same query -> *failed to allocate data of size 2.0MB (9.7MB/10.0MB used)* -> ok, so we need some RAM, but how much do we need?
* concatenate 4 files into 1GB parquet file (x2 RAM) -> run the query -> it works! 

Confused about supposed to work and when...

**Comment by arjenpdevries** [comment]
The _working set_ must fit in RAM :-)

But a bit hard to estimate without running it first.
